### PR TITLE
AQ32: hal support

### PR DESCRIPTION
**Pull request by mlyle**
Changes to other targets because of HAL change for PWM support.

d-ronin/dronin#17

### CODE REVIEW

**Comment by kubark42** [comment]
Can someone confirm that this works? I don't have an AQ32 for testing.

**Comment by jihlein** [comment]
Already have tested it (on the bench).  Port's configure properly.

Date: Wed, 14 Oct 2015 12:48:53 -0700

File: flight/PiOS/Common/pios_hal.c [hunk 1]
```diff
@@ -368,7 +370,7 @@ void PIOS_HAL_ConfigurePort(HwSharedPortTypesOptions port_type,
 	uintptr_t port_driver_id;
 	uintptr_t *target=NULL, *target2=NULL;;
 
-        switch (port_type) {
+    switch (port_type) {
```
Comment (kubark42) [inline flight/PiOS/Common/pios_hal.c]: Inconsistent indentation.

File: flight/PiOS/Common/pios_hal.c [hunk 2]
```diff
@@ -512,7 +530,7 @@ void PIOS_HAL_ConfigurePort(HwSharedPortTypesOptions port_type,
                 GPIO_Init(sbus_cfg->inv.gpio, (GPIO_InitTypeDef*)&sbus_cfg->inv.init);
                 GPIO_WriteBit(sbus_cfg->inv.gpio, sbus_cfg->inv.init.GPIO_Pin, sbus_cfg->gpio_inv_disable);
         }
-
+
```
Comment (kubark42) [inline flight/PiOS/Common/pios_hal.c]: Get rid of the whitespace in this newline.

File: flight/targets/aq32/board-info/board_hw_defs.c [hunk 3]
```diff
@@ -911,10 +913,9 @@ static const struct pios_sbus_cfg pios_usart3_sbus_aux_cfg = {
 		},
 	},
 	.gpio_inv_enable = Bit_SET,
+	.gpio_inv_disable = Bit_RESET,
 };
 
-#endif	/* PIOS_INCLUDE_SBUS */
```
Comment (kubark42) [inline flight/targets/aq32/board-info/board_hw_defs.c]: Why was the #endif moved? It looks like it affects an SBUS configuration.

**Comment by kubark42** [comment]
LGTM. Fix the whitespace issues and it can merge. 

Can you also check that this does not apply: https://github.com/TauLabs/TauLabs/issues/1926?

File: flight/targets/aq32/board-info/board_hw_defs.c (hunk 3 above)
Comment (mlyle) [inline flight/targets/aq32/board-info/board_hw_defs.c]: Will move it back; I'm betting it was accidental.

Not a fan of these in the board_hw_defs, etc.  If the object isn't used the linker will omit it, so they're non-functional.

**Comment by mlyle** [comment]
I don't have AQ32 but @jihlein indicated that a spot-check of functionality across ports was good.

**Comment by jihlein** [comment]
@mlyle, did you do the whitespace corrections?

**Comment by mlyle** [comment]
Yes.  I ran it through uncrustify to normalize the styling.

**Comment by kubark42** [comment]
I really liked @peabody124's suggestion for placing uncrustify commands in the commit message. @mlyle, it's not important for this one, but in future commits could you add the uncrustify command per James's suggestion? I like that we're improving code quality, but I'm a little lost in what's being done and how to do it so I get identical results for my PRs.

> @jihlein indicated that a spot-check of functionality across ports was good.

Huh, I wonder what I did wrong on my internal board stuff. If another board comes over and doesn't have UAVO browser issues either, then I think we should close #1926.

**Comment by kubark42** [comment]
P.S. @mlyle I'll merge this as is, but let me know if you'd like time to rebase to change the uncrustify commit message. As I said, not a blocker, just thought I'd give you the chance if you wanted it.

**Comment by jihlein** [comment]
Can this be merged?  I'd like to submit a PR for the AQ32 OpenLog integration, but need this merged first.

**Comment by mlyle** [comment]
The uncrustify stuff is now in its own PR which can now form the documentation of how it was done.

**Comment by lilvinz** [comment]
jenkins: test this please

**Comment by kubark42** [comment]
Thanks for the update, @mlyle.

@jihlein, sorry for the delay, I was sailing all weekend and hadn't turned on my computer. Looking forward to your OpenLog PR.

**Comment by jihlein** [comment]
@kubark42, thanks!

**Pull request merged**